File: sdk/typescript/examples/basic.ts
/**
 * Basic usage of the HyperMesh TypeScript SDK.
 *
 * Run: npx tsx examples/basic.ts
 */
import { HyperMeshClient, HyperMeshError } from "../src/index.js";

async function main() {
  const client = new HyperMeshClient(); // http://localhost:9293

  // Ping the node
  const pong = await client.node.ping();
  console.log("Ping:", pong);

  // Get node status
  const status = await client.node.status();
  console.log("Node ID:", status.node_id);
  console.log("Chain height:", status.chain_height);
  console.log("Peers:", status.peers);
  console.log("Privacy mode:", status.privacy_mode);
  console.log("Uptime:", status.uptime_secs, "seconds");
  console.log("Coordinate:", status.coordinate);

  // List connected peers
  const peers = await client.network.peers();
  console.log("\nConnected peers:", peers.count);
  for (const peer of peers.peers) {
    console.log(" -", JSON.stringify(peer));
  }

  // Error handling
  try {
    await client.blockchain.block(999999);
  } catch (err) {
    if (err instanceof HyperMeshError) {
      console.log("\nExpected error for block 999999:");
      console.log("  Status:", err.status);
      console.log("  Message:", err.message);
    }
  }
}

main().catch(console.error);

File: sdk/python/examples/basic.py
"""Basic usage of the HyperMesh Python SDK.

Run: python examples/basic.py
"""

from hypermesh import HyperMeshClient, HyperMeshError, NotFoundError

def main() -> None:
    client = HyperMeshClient()  # https://localhost:8443

    # Ping the node
    alive = client.node.ping()
    print(f"Node alive: {alive}")

    # Get node status
    status = client.node.status()
    print(f"Node ID: {status.node_id}")
    print(f"Chain height: {status.chain_height}")
    print(f"Peers: {status.peers}")
    print(f"Privacy mode: {status.privacy_mode}")
    print(f"Uptime: {status.uptime_secs} seconds")
    print(f"Coordinate: {status.coordinate}")

    # List connected peers
    peers = client.network.peers()
    print(f"\nConnected peers: {peers.count}")
    for peer in peers.peers:
        print(f"  {peer.node_id} @ {peer.address}")

    # Error handling
    try:
        client.blockchain.block(999999)
    except NotFoundError:
        print("\nExpected: block 999999 not found")
    except HyperMeshError as e:
        print(f"\nError fetching block 999999: {e}")


if __name__ == "__main__":
    main()
